### PR TITLE
Use http Stream for describe ring

### LINKED ISSUE #6297: oversized allocation during manager repair
*Installation details*
Scylla version (or git commit hash):666.development-0.20200423.fbcf741c2 with build-id cdcc3451a8e1a80d81a647038c0c28c823345d8d
Cluster size:6
OS (RHEL/CentOS/Ubuntu/AWS AMI):ami-0681759688f9cbe67(Ireland)

On the 4 hours 100 GB longevity, during a manager repair, all of the living nodes experienced a seastar_memory oversized allocation issue:

node#7:
```
2020-04-23T23:27:09+00:00  longevity-100gb-4h-master-db-node-5ff1d062-7 !WARNING | scylla: [shard 0] seastar_memory - oversized allocation: 1052672 bytes. This is non-fatal, but could lead to latency and/or fragmentation issues. Please report: at   0x324d59d#012  0x2d7e2a4#012  0x2d82fab#012  0x7f72cf8d96f1#012  0x7f72ce21b32a#012  0x7f72ce2271c9#012  0x7f72ce215936#012  0x27169ea#012  0x26c5fd3#012  0x30b3c2e#012  0x30b6212#012  0x30bca16#012  0x309c60f#012  0x30a0d17#012  0x30a324d#012  0x2dc1e60#012  0x2dc206f#012  0x2df77f5#012  0x2d61a21#012  0x2d62b6e#012  0xd0bea7#012  /opt/scylladb/libreloc/libc.so.6+0x271a2#012  0xc2d02d
```

Translated backtrace:
```
backtrace<seastar::current_backtrace()::<lambda(seastar::frame)> > at /usr/include/fmt/format.h:1158
 (inlined by) seastar::current_backtrace() at ./build/release/seastar/./seastar/src/util/backtrace.cc:84
seastar::memory::cpu_pages::warn_large_allocation(unsigned long) at /usr/include/c++/9/ostream:221
seastar::memory::cpu_pages::check_large_allocation(unsigned long) at /usr/include/c++/9/ostream:221
 (inlined by) seastar::memory::cpu_pages::allocate_large(unsigned int) at ./build/release/seastar/./seastar/src/core/memory.cc:677
 (inlined by) seastar::memory::allocate_large(unsigned long) at ./build/release/seastar/./seastar/src/core/memory.cc:1216
 (inlined by) seastar::memory::allocate(unsigned long) at ./build/release/seastar/./seastar/src/core/memory.cc:1268
 (inlined by) operator new(unsigned long) at ./build/release/seastar/./seastar/src/core/memory.cc:1694
?? ??:0
?? ??:0
?? ??:0
?? ??:0
std::basic_ostream<char, std::char_traits<char> >& seastar::operator<< <char, unsigned int, 15u, true, std::char_traits<char> >(std::basic_ostream<char, std::char_traits<char> >&, seastar::basic_sstring<char, unsigned int, 15u, true> const&) at /usr/include/fmt/format.h:1158
 (inlined by) seastar::basic_sstring<char, unsigned int, 15u, true> seastar::json::formatter::to_json<__gnu_cxx::__normal_iterator<seastar::httpd::storage_service_json::token_range const*, std::vector<seastar::httpd::storage_service_json::token_range, std::allocator<seastar::httpd::storage_service_json::token_range> > > >(seastar::json::formatter::state, __gnu_cxx::__normal_iterator<seastar::httpd::storage_service_json::token_range const*, std::vector<seastar::httpd::storage_service_json::token_range, std::allocator<seastar::httpd::storage_service_json::token_range> > >, __gnu_cxx::__normal_iterator<seastar::httpd::storage_service_json::token_range const*, std::vector<seastar::httpd::storage_service_json::token_range, std::allocator<seastar::httpd::storage_service_json::token_range> > >) at ././seastar/include/seastar/json/formatter.hh:69
seastar::basic_sstring<char, unsigned int, 15u, true> seastar::json::formatter::to_json<seastar::httpd::storage_service_json::token_range, std::allocator<seastar::httpd::storage_service_json::token_range> >(std::vector<seastar::httpd::storage_service_json::token_range, std::allocator<seastar::httpd::storage_service_json::token_range> > const&) at /usr/include/fmt/format.h:1158
 (inlined by) seastar::json::json_return_type::json_return_type<std::vector<seastar::httpd::storage_service_json::token_range, std::allocator<seastar::httpd::storage_service_json::token_range> > >(std::vector<seastar::httpd::storage_service_json::token_range, std::allocator<seastar::httpd::storage_service_json::token_range> > const&) at ././seastar/include/seastar/json/json_elements.hh:292
 (inlined by) _M_invoke at /usr/include/c++/9/bits/std_function.h:286
std::function<seastar::json::json_return_type (seastar::httpd::request const&)>::operator()(seastar::httpd::request const&) const at ./build/release/seastar/./seastar/include/seastar/core/future.hh:363
 (inlined by) seastar::httpd::function_handler::function_handler(std::function<seastar::json::json_return_type (seastar::httpd::request const&)> const&)::{lambda(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)#1}::operator()(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >) const at ./build/release/seastar/./seastar/include/seastar/http/function_handlers.hh:93
 (inlined by) std::_Function_handler<seastar::future<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > > (std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >), seastar::httpd::function_handler::function_handler(std::function<seastar::json::json_return_type (seastar::httpd::request const&)> const&)::{lambda(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)#1}>::_M_invoke(std::_Any_data const&, std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >&&, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >&&) at /usr/include/c++/9/bits/std_function.h:286
std::function<seastar::future<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > > (std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)>::operator()(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >) const at /usr/include/c++/9/bits/std_function.h:688
 (inlined by) seastar::httpd::function_handler::handle(seastar::basic_sstring<char, unsigned int, 15u, true> const&, std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >) at ./build/release/seastar/./seastar/include/seastar/http/function_handlers.hh:118
seastar::httpd::routes::handle(seastar::basic_sstring<char, unsigned int, 15u, true> const&, std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >) at /usr/include/c++/9/sstream:65
seastar::httpd::connection::generate_reply(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >) at /usr/include/fmt/format.h:1158
operator() at /usr/include/fmt/format.h:1158
 (inlined by) apply at ./build/release/seastar/./seastar/include/seastar/core/apply.hh:36
 (inlined by) apply<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)>::<lambda()> > at ./build/release/seastar/./seastar/include/seastar/core/apply.hh:44
 (inlined by) apply<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)>::<lambda()> > at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1682
 (inlined by) then_impl<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)>::<lambda()> > at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1250
 (inlined by) then<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)>::<lambda()> > at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1210
 (inlined by) operator() at ./build/release/seastar/./seastar/src/http/httpd.cc:239
 (inlined by) apply at ./build/release/seastar/./seastar/include/seastar/core/apply.hh:36
 (inlined by) apply<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)>, std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> > > at ./build/release/seastar/./seastar/include/seastar/core/apply.hh:44
 (inlined by) apply<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)>, std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> > > at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1682
 (inlined by) then_impl<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)> > at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1250
 (inlined by) then<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)> > at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1210
 (inlined by) operator() at ./build/release/seastar/./seastar/src/http/httpd.cc:244
 (inlined by) apply at ./build/release/seastar/./seastar/include/seastar/core/apply.hh:36
 (inlined by) apply<seastar::httpd::connection::read_one()::<lambda()> > at ./build/release/seastar/./seastar/include/seastar/core/apply.hh:44
 (inlined by) apply<seastar::httpd::connection::read_one()::<lambda()> > at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1682
operator() at /usr/include/fmt/format.h:1158
 (inlined by) run_and_dispose at ./build/release/seastar/./seastar/include/seastar/core/future.hh:515
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at /usr/include/fmt/format.h:1158
seastar::reactor::run_some_tasks() at /usr/include/fmt/format.h:1158
seastar::reactor::run_some_tasks() at /usr/include/fmt/format.h:1158
 (inlined by) seastar::reactor::run() at ./build/release/seastar/./seastar/src/core/reactor.cc:2709
seastar::app_template::run_deprecated(int, char**, std::function<void ()>&&) at /usr/include/boost/program_options/variables_map.hpp:146
seastar::app_template::run(int, char**, std::function<seastar::future<int> ()>&&) at /usr/include/boost/program_options/variables_map.hpp:146
main at ./main.cc:483
?? ??:0
_start at ??:?
```

Node#1:
```
2020-04-23T23:27:09+00:00  longevity-100gb-4h-master-db-node-5ff1d062-1 !WARNING | scylla: [shard 0] seastar_memory - oversized allocation: 1052672 bytes. This is non-fatal, but could lead to latency and/or fragmentation issues. Please report: at   0x324d59d#012  0x2d7e2a4#012  0x2d82fab#012  0x7fbdf5dee6f1#012  0x7fbdf473032a#012  0x7fbdf473c1c9#012  0x7fbdf472a936#012  0x27169ea#012  0x26c5fd3#012  0x30b3c2e#012  0x30b6212#012  0x30bca16#012  0x309c60f#012  0x30a0d17#012  0x30a324d#012  0x2dc1e60#012  0x2dc206f#012  0x2df77f5#012  0x2d61a21#012  0x2d62b6e#012  0xd0bea7#012  /opt/scylladb/libreloc/libc.so.6+0x271a2#012  0xc2d02d
```

Translated backtrace:
```
backtrace<seastar::current_backtrace()::<lambda(seastar::frame)> > at /usr/include/fmt/format.h:1158
 (inlined by) seastar::current_backtrace() at ./build/release/seastar/./seastar/src/util/backtrace.cc:84
seastar::memory::cpu_pages::warn_large_allocation(unsigned long) at /usr/include/c++/9/ostream:221
seastar::memory::cpu_pages::check_large_allocation(unsigned long) at /usr/include/c++/9/ostream:221
 (inlined by) seastar::memory::cpu_pages::allocate_large(unsigned int) at ./build/release/seastar/./seastar/src/core/memory.cc:677
 (inlined by) seastar::memory::allocate_large(unsigned long) at ./build/release/seastar/./seastar/src/core/memory.cc:1216
 (inlined by) seastar::memory::allocate(unsigned long) at ./build/release/seastar/./seastar/src/core/memory.cc:1268
 (inlined by) operator new(unsigned long) at ./build/release/seastar/./seastar/src/core/memory.cc:1694
?? ??:0
?? ??:0
?? ??:0
?? ??:0
std::basic_ostream<char, std::char_traits<char> >& seastar::operator<< <char, unsigned int, 15u, true, std::char_traits<char> >(std::basic_ostream<char, std::char_traits<char> >&, seastar::basic_sstring<char, unsigned int, 15u, true> const&) at /usr/include/fmt/format.h:1158
 (inlined by) seastar::basic_sstring<char, unsigned int, 15u, true> seastar::json::formatter::to_json<__gnu_cxx::__normal_iterator<seastar::httpd::storage_service_json::token_range const*, std::vector<seastar::httpd::storage_service_json::token_range, std::allocator<seastar::httpd::storage_service_json::token_range> > > >(seastar::json::formatter::state, __gnu_cxx::__normal_iterator<seastar::httpd::storage_service_json::token_range const*, std::vector<seastar::httpd::storage_service_json::token_range, std::allocator<seastar::httpd::storage_service_json::token_range> > >, __gnu_cxx::__normal_iterator<seastar::httpd::storage_service_json::token_range const*, std::vector<seastar::httpd::storage_service_json::token_range, std::allocator<seastar::httpd::storage_service_json::token_range> > >) at ././seastar/include/seastar/json/formatter.hh:69
seastar::basic_sstring<char, unsigned int, 15u, true> seastar::json::formatter::to_json<seastar::httpd::storage_service_json::token_range, std::allocator<seastar::httpd::storage_service_json::token_range> >(std::vector<seastar::httpd::storage_service_json::token_range, std::allocator<seastar::httpd::storage_service_json::token_range> > const&) at /usr/include/fmt/format.h:1158
 (inlined by) seastar::json::json_return_type::json_return_type<std::vector<seastar::httpd::storage_service_json::token_range, std::allocator<seastar::httpd::storage_service_json::token_range> > >(std::vector<seastar::httpd::storage_service_json::token_range, std::allocator<seastar::httpd::storage_service_json::token_range> > const&) at ././seastar/include/seastar/json/json_elements.hh:292
 (inlined by) _M_invoke at /usr/include/c++/9/bits/std_function.h:286
std::function<seastar::json::json_return_type (seastar::httpd::request const&)>::operator()(seastar::httpd::request const&) const at ./build/release/seastar/./seastar/include/seastar/core/future.hh:363
 (inlined by) seastar::httpd::function_handler::function_handler(std::function<seastar::json::json_return_type (seastar::httpd::request const&)> const&)::{lambda(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)#1}::operator()(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >) const at ./build/release/seastar/./seastar/include/seastar/http/function_handlers.hh:93
 (inlined by) std::_Function_handler<seastar::future<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > > (std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >), seastar::httpd::function_handler::function_handler(std::function<seastar::json::json_return_type (seastar::httpd::request const&)> const&)::{lambda(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)#1}>::_M_invoke(std::_Any_data const&, std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >&&, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >&&) at /usr/include/c++/9/bits/std_function.h:286
std::function<seastar::future<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > > (std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)>::operator()(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >) const at /usr/include/c++/9/bits/std_function.h:688
 (inlined by) seastar::httpd::function_handler::handle(seastar::basic_sstring<char, unsigned int, 15u, true> const&, std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >) at ./build/release/seastar/./seastar/include/seastar/http/function_handlers.hh:118
seastar::httpd::routes::handle(seastar::basic_sstring<char, unsigned int, 15u, true> const&, std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >) at /usr/include/c++/9/sstream:65
seastar::httpd::connection::generate_reply(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >) at /usr/include/fmt/format.h:1158
operator() at /usr/include/fmt/format.h:1158
 (inlined by) apply at ./build/release/seastar/./seastar/include/seastar/core/apply.hh:36
 (inlined by) apply<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)>::<lambda()> > at ./build/release/seastar/./seastar/include/seastar/core/apply.hh:44
 (inlined by) apply<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)>::<lambda()> > at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1682
 (inlined by) then_impl<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)>::<lambda()> > at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1250
 (inlined by) then<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)>::<lambda()> > at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1210
 (inlined by) operator() at ./build/release/seastar/./seastar/src/http/httpd.cc:239
 (inlined by) apply at ./build/release/seastar/./seastar/include/seastar/core/apply.hh:36
 (inlined by) apply<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)>, std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> > > at ./build/release/seastar/./seastar/include/seastar/core/apply.hh:44
 (inlined by) apply<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)>, std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> > > at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1682
 (inlined by) then_impl<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)> > at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1250
 (inlined by) then<seastar::httpd::connection::read_one()::<lambda()> mutable::<lambda(std::unique_ptr<seastar::httpd::request>)> > at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1210
 (inlined by) operator() at ./build/release/seastar/./seastar/src/http/httpd.cc:244
 (inlined by) apply at ./build/release/seastar/./seastar/include/seastar/core/apply.hh:36
 (inlined by) apply<seastar::httpd::connection::read_one()::<lambda()> > at ./build/release/seastar/./seastar/include/seastar/core/apply.hh:44
 (inlined by) apply<seastar::httpd::connection::read_one()::<lambda()> > at ./build/release/seastar/./seastar/include/seastar/core/future.hh:1682
operator() at /usr/include/fmt/format.h:1158
 (inlined by) run_and_dispose at ./build/release/seastar/./seastar/include/seastar/core/future.hh:515
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at /usr/include/fmt/format.h:1158
seastar::reactor::run_some_tasks() at /usr/include/fmt/format.h:1158
seastar::reactor::run_some_tasks() at /usr/include/fmt/format.h:1158
 (inlined by) seastar::reactor::run() at ./build/release/seastar/./seastar/src/core/reactor.cc:2709
seastar::app_template::run_deprecated(int, char**, std::function<void ()>&&) at /usr/include/boost/program_options/variables_map.hpp:146
seastar::app_template::run(int, char**, std::function<seastar::future<int> ()>&&) at /usr/include/boost/program_options/variables_map.hpp:146
main at ./main.cc:483
?? ??:0
_start at ??:?
```
The rest of the nodes seem to experience the issue with a very similar, if no identical, backtrace:
node#2
```
2020-04-23T23:26:39+00:00  longevity-100gb-4h-master-db-node-5ff1d062-2 !WARNING | scylla: [shard 0] seastar_memory - oversized allocation: 1052672 bytes. This is non-fatal, but could lead to latency and/or fragmentation issues. Please report: at   0x324d59d#012  0x2d7e2a4#012  0x2d82fab#012  0x7f3ced53c6f1#012  0x7f3cebe7e32a#012  0x7f3cebe8a1c9#012  0x7f3cebe78936#012  0x27169ea#012  0x26c5fd3#012  0x30b3c2e#012  0x30b6212#012  0x30bca16#012  0x309c60f#012  0x30a0d17#012  0x30a324d#012  0x2dc1e60#012  0x2dc206f#012  0x2df77f5#012  0x2d61a21#012  0x2d62b6e#012  0xd0bea7#012  /opt/scylladb/libreloc/libc.so.6+0x271a2#012  0xc2d02d
```
node#3:
```
2020-04-23T23:26:39+00:00  longevity-100gb-4h-master-db-node-5ff1d062-3 !WARNING | scylla: [shard 0] seastar_memory - oversized allocation: 1052672 bytes. This is non-fatal, but could lead to latency and/or fragmentation issues. Please report: at   0x324d59d#012  0x2d7e2a4#012  0x2d82fab#012  0x7f6784bdc6f1#012  0x7f678351e32a#012  0x7f678352a1c9#012  0x7f6783518936#012  0x27169ea#012  0x26c5fd3#012  0x30b3c2e#012  0x30b6212#012  0x30bca16#012  0x309c60f#012  0x30a0d17#012  0x30a324d#012  0x2dc1e60#012  0x2dc206f#012  0x2df77f5#012  0x2d61a21#012  0x2d62b6e#012  0xd0bea7#012  /opt/scylladb/libreloc/libc.so.6+0x271a2#012  0xc2d02d
```
node#8:
```
2020-04-23T23:26:39+00:00  longevity-100gb-4h-master-db-node-5ff1d062-8 !WARNING | scylla: [shard 0] seastar_memory - oversized allocation: 1052672 bytes. This is non-fatal, but could lead to latency and/or fragmentation issues. Please report: at   0x324d59d#012  0x2d7e2a4#012  0x2d82fab#012  0x7f8058a4f6f1#012  0x7f805739132a#012  0x7f805739d1c9#012  0x7f805738b936#012  0x27169ea#012  0x26c5fd3#012  0x30b3c2e#012  0x30b6212#012  0x30bca16#012  0x309c60f#012  0x30a0d17#012  0x30a324d#012  0x2dc1e60#012  0x2dc206f#012  0x2df77f5#012  0x2d61a21#012  0x2d62b6e#012  0xd0bea7#012  /opt/scylladb/libreloc/libc.so.6+0x271a2#012  0xc2d02d
```
node#9:
```
2020-04-23T23:26:39+00:00  longevity-100gb-4h-master-db-node-5ff1d062-9 !WARNING | scylla: [shard 0] seastar_memory - oversized allocation: 1052672 bytes. This is non-fatal, but could lead to latency and/or fragmentation issues. Please report: at   0x324d59d#012  0x2d7e2a4#012  0x2d82fab#012  0x7f7a7009c6f1#012  0x7f7a6e9de32a#012  0x7f7a6e9ea1c9#012  0x7f7a6e9d8936#012  0x27169ea#012  0x26c5fd3#012  0x30b3c2e#012  0x30b6212#012  0x30bca16#012  0x309c60f#012  0x30a0d17#012  0x30a324d#012  0x2dc1e60#012  0x2dc206f#012  0x2df77f5#012  0x2d61a21#012  0x2d62b6e#012  0xd0bea7#012  /opt/scylladb/libreloc/libc.so.6+0x271a2#012  0xc2d02d
```

This error caused the connection with two of the nodes, node#1 and node#7, to break due to a broken pipe:
```
2020-04-23 23:27:08.000: (DatabaseLogEvent Severity.WARNING): type=BROKEN_PIPE regex=cql_server - exception while processing connection:.*Broken pipe line_number=88603 node=Node longevity-100gb-4h-master-db-node-5ff1d062-7 [54.229.39.63 | 10.0.32.67] (seed: False)
2020-04-23T23:27:08+00:00  longevity-100gb-4h-master-db-node-5ff1d062-7 !INFO    | scylla: [shard 13] cql_server - exception while processing connection: std::system_error (error system:32, sendmsg: Broken pipe)

_start at ??:?
20-04-23 23:27:08.000: (DatabaseLogEvent Severity.WARNING): type=BROKEN_PIPE regex=cql_server - exception while processing connection:.*Broken pipe line_number=86723 node=Node longevity-100gb-4h-master-db-node-5ff1d062-1 [52.209.152.170 | 10.0.48.71] (seed: True)
020-04-23T23:27:08+00:00  longevity-100gb-4h-master-db-node-5ff1d062-1 !INFO    | scylla: [shard 0] cql_server - exception while processing connection: std::system_error (error system:32, sendmsg: Broken pipe)
```


node list:
```
longevity-100gb-4h-master-db-node-5ff1d062-1 [52.209.152.170 | 10.0.48.71]
longevity-100gb-4h-master-db-node-5ff1d062-2 [54.229.96.96 | 10.0.121.248]
longevity-100gb-4h-master-db-node-5ff1d062-3 [34.246.178.161 | 10.0.91.42]
longevity-100gb-4h-master-db-node-5ff1d062-4 [34.242.85.112 | 10.0.223.93]
longevity-100gb-4h-master-db-node-5ff1d062-5 [34.243.32.130 | 10.0.37.174]
longevity-100gb-4h-master-db-node-5ff1d062-6 [3.250.47.238 | 10.0.187.86]
longevity-100gb-4h-master-db-node-5ff1d062-7 [54.229.39.63 | 10.0.32.67]
longevity-100gb-4h-master-db-node-5ff1d062-8 [54.154.209.217 | 10.0.51.210]
longevity-100gb-4h-master-db-node-5ff1d062-9 [54.76.160.182 | 10.0.31.85]
```

The job's logs:
```
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                            Log links for testrun with test id 5ff1d062-e6b2-4e61-bbba-d8c6fe2d1589                                                                                             |
+-----------------+-------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Date            | Log type    | Link                                                                                                                                                                                                                           |
+-----------------+-------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 20190101_010101 | prometheus  | https://cloudius-jenkins-test.s3.amazonaws.com/5ff1d062-e6b2-4e61-bbba-d8c6fe2d1589/prometheus_snapshot_20200423_233830.tar.gz                                                                                                 |
| 20200423_233450 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/5ff1d062-e6b2-4e61-bbba-d8c6fe2d1589/20200423_233450/grafana-screenshot-overview-20200423_233637-longevity-100gb-4h-master-monitor-node-5ff1d062-1.png                          |
| 20200423_233450 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/5ff1d062-e6b2-4e61-bbba-d8c6fe2d1589/20200423_233450/grafana-screenshot-scylla-per-server-metrics-nemesis-20200423_233452-longevity-100gb-4h-master-monitor-node-5ff1d062-1.png |
| 20200423_234252 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/5ff1d062-e6b2-4e61-bbba-d8c6fe2d1589/20200423_234252/grafana-screenshot-overview-20200423_234425-longevity-100gb-4h-master-monitor-node-5ff1d062-1.png                          |
| 20200423_234252 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/5ff1d062-e6b2-4e61-bbba-d8c6fe2d1589/20200423_234252/grafana-screenshot-scylla-per-server-metrics-nemesis-20200423_234253-longevity-100gb-4h-master-monitor-node-5ff1d062-1.png |
| 20200423_234915 | db-cluster  | https://cloudius-jenkins-test.s3.amazonaws.com/5ff1d062-e6b2-4e61-bbba-d8c6fe2d1589/20200423_234915/db-cluster-5ff1d062.zip                                                                                                    |
| 20200423_234915 | loader-set  | https://cloudius-jenkins-test.s3.amazonaws.com/5ff1d062-e6b2-4e61-bbba-d8c6fe2d1589/20200423_234915/loader-set-5ff1d062.zip                                                                                                    |
| 20200423_234915 | monitor-set | https://cloudius-jenkins-test.s3.amazonaws.com/5ff1d062-e6b2-4e61-bbba-d8c6fe2d1589/20200423_234915/monitor-set-5ff1d062.zip                                                                                                   |
| 20200423_234915 | sct-runner  | https://cloudius-jenkins-test.s3.amazonaws.com/5ff1d062-e6b2-4e61-bbba-d8c6fe2d1589/20200423_234915/sct-runner-5ff1d062.zip                                                                                                    |
+-----------------+-------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

```

### PR DESCRIPTION
**Pull request by amnonh**
This series changes the describe_ring API to use HTTP stream instead of serializing the results and send it as a single buffer.

While testing the change I hit a 4-year-old issue inside service/storage_proxy.cc that causes a use after free, so I fixed it along the way.

Fixes #6297

### CODE REVIEW

File: service/storage_service.cc [hunk 1]
```diff
@@ -1040,13 +1040,17 @@ storage_service::is_local_dc(const inet_address& targetHost) const {
...
+    }
     // some people just want to get a visual representation of things. Allow null and set it to the first
     // non-system keyspace.
-    if (keyspace == "" && _db.local().get_non_system_keyspaces().empty()) {
+    auto keyspaces = _db.local().get_non_system_keyspaces();
+
+    if (keyspaces.empty()) {
         throw std::runtime_error("No keyspace provided and no non system kespace exist");
     }
-    const sstring& ks = (keyspace == "") ? _db.local().get_non_system_keyspaces()[0] : keyspace;
```
Comment (avikivity) [inline service/storage_service.cc:1048]: Please file an issue for this and add a "Fixes".

File: service/storage_service.cc [hunk 2]
```diff
@@ -1040,13 +1040,17 @@ storage_service::is_local_dc(const inet_address& targetHost) const {
...
-    return construct_range_to_endpoint_map(ks, get_all_ranges(sorted_tokens));
+    return construct_range_to_endpoint_map(keyspaces[0], get_all_ranges(sorted_tokens));
```
Comment (avikivity) [inline service/storage_service.cc:1053]: It's nicer to have just one call, it allows the reader to verify that the only difference is how we get keyspace.

```c++
   if (keyspace.empty()) }
           // blah blah, check whatever
           keyspace = ...;
   }
   return construct_range_to_endpoint_map(keyspace, ...)
```

**Comment by avikivity** [comment]
Looks good, I left a cosmetic comment.

File: service/storage_service.cc (hunk 2 above)
Comment (amnonh) [reply service/storage_service.cc:1053]: I guess that was the original thought behind the original implementation.
`keyspace` is a const reference. it's not a critical path and it doesn't happen too often, so  I can use a string by value.
If you are ok with it, I'll change the implementation accordingly.

Comment (avikivity) [reply service/storage_service.cc:1053]: String by value is fine here.

**Pull request merged**